cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx

```diff
@@ -15,61 +15,65 @@
  */
 
 import React from 'react';
-import Table from '@material-ui/core/Table';
-import TableBody from '@material-ui/core/TableBody';
-import TableHead from '@material-ui/core/TableHead';
-import TableRow from '@material-ui/core/TableRow';
 import Paper from '@material-ui/core/Paper';
-import TableCell from '@material-ui/core/TableCell';
-import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+import Grid from '@material-ui/core/Grid';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
 import isEmpty from 'lodash/isEmpty';
 import { PREVIEW_STATUS } from 'services/PreviewStatus';
 import Heading, { HeadingTypes } from 'components/Heading';
 import ThemeWrapper from 'components/ThemeWrapper';
+import VirtualScroll from 'components/VirtualScroll';
 import T from 'i18n-react';
+import classnames from 'classnames';
 
 const I18N_PREFIX = 'features.PreviewData.DataView.Table';
 
-export const CustomTableCell = withStyles((theme) => ({
-  head: {
-    backgroundColor: theme.palette.grey['300'],
-    color: theme.palette.common.white,
-    padding: 10,
-    fontSize: 14,
-    '&:first-of-type': {
-      borderRight: `1px solid ${theme.palette.grey['500']}`,
-    },
-  },
-  body: {
-    padding: 10,
-    fontSize: 14,
-    '&:first-of-type': {
-      borderRight: `1px solid ${theme.palette.grey['500']}`,
-    },
-  },
-}))(TableCell);
-
 export const messageTextStyle = {
   fontSize: '1.3rem !important',
-  margin: '10px 0',
+  margin: 'unset',
+  padding: '10px 5px',
 };
-export const styles = (theme) => ({
+export const styles = (theme): StyleRules => ({
   root: {
-    width: '100%',
     display: 'inline-block',
     height: 'auto',
     marginTop: theme.spacing(1),
   },
-  table: {
-    width: '100%',
+  tableContainer: {
+    width: 'fit-content',
+    minWidth: '100%',
   },
   row: {
     height: 40,
-    '&:nth-of-type(odd)': {
+    '&.oddRow': {
```

Comment:
Curious: Why switch from the pseudo-class to an explicit class?

Reply:
Nm, looks like the virtual scroll would alter the oddness/evenness of a given row.

Reply:
Great question - I did this so the color of a row didn't change/flicker randomly now that we're dynamically generating the rows with virtual scroll.